python-checks/src/main/java/org/sonar/python/checks/BuiltinShadowingAssignmentCheck.java

```diff
@@ -86,15 +93,35 @@ private void checkAnnotatedAssignment(SubscriptionContext ctx) {
   }
 
   private void raiseIssueForNonGlobalVariable(SubscriptionContext ctx, Name variable) {
-    Symbol variableSymbol = variable.symbol();
-    if (variableSymbol != null && variableSymbol.usages().stream().noneMatch(u -> u.kind().equals(Usage.Kind.GLOBAL_DECLARATION))) {
-      PreciseIssue existingIssue = variableIssuesRaised.get(variableSymbol);
-      if (existingIssue != null) {
-        existingIssue.secondary(variable, REPEATED_VAR_MESSAGE);
-      } else {
-        variableIssuesRaised.put(variableSymbol, ctx.addIssue(variable, MESSAGE));
-      }
-    }
+    Optional.ofNullable(variable.symbol())
+      .filter(symbol -> symbol.usages().stream().map(Usage::kind).noneMatch(Usage.Kind.GLOBAL_DECLARATION::equals))
+      .ifPresent(symbol -> {
+        var existingIssue = variableIssuesRaised.get(symbol);
+        if (existingIssue != null) {
+          existingIssue.secondary(variable, REPEATED_VAR_MESSAGE);
+        } else {
+          var issue = (IssueWithQuickFix) ctx.addIssue(variable, MESSAGE);
+
+          PythonQuickFix quickFix = createQuickFix(symbol);
+          issue.addQuickFix(quickFix);
+          variableIssuesRaised.put(symbol, issue);
+        }
+      });
+  }
+
+  private static PythonQuickFix createQuickFix(Symbol symbol) {
+    var edits = symbol.usages()
+      .stream()
+      .map(Usage::tree)
+      .map(Tree::firstToken)
+      .sorted(Comparator.comparing(Token::line).reversed().thenComparing(Token::column).reversed())
+      .map(token -> PythonTextEdit.insertBefore(token, "_"))
```

Comment:
What if another variable already uses that name?
```
def foo():
  max = 1
  _max = 2
  print(max)
```
I know, it's probably a corner case.
If it's not easy to fix, maybe we can create a separate ticket to handle it later.

Reply:
Some protection was added to prevent renaming to already used value. It will not help in 100% of cases. If you have any ideas on how to make it in a more proper way - let's discuss.